experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -319,7 +319,7 @@ export interface MapNode<TSchema extends MapSchema> extends TreeNode {
 	 */
 	[boxedIterator](): IterableIterator<TypedField<TSchema["mapFields"]>>;
 
-	[Symbol.iterator](): IterableIterator<UnboxField<TSchema["mapFields"], "notEmpty">>;
+	[Symbol.iterator](): IterableIterator<[FieldKey, UnboxField<TSchema["mapFields"], "notEmpty">]>;
```

Comment:
Since this does unboxing now providing the key here is a good change.

Reply:
Good point, I hadn't actually considered that. The key isn't necessarily implicitly included after unboxing like it is when it stays boxed.